ChangeLog

```diff
@@ -3,12 +3,66 @@ Changelog
 
 ## v2.4.0
 
-- Enum: uniq is modified to use structural comparison internally, instead
-    of previously used physical comparison, which was most likely a bug.
-    The old behavior is provided by the new Enum.uniqq function.
-    Enum.uniq_by was added for additional convenience.
+- preparation for upcoming '-safe-string compat' OCaml warning
```

Comment:
I find this line hard to understand. I tried to propose a reformulation and failed to find something convincing. Could we just drop it?